### PR TITLE
Glossary, refs, and stubs

### PR DESCRIPTION
**Pull request by collijk**
- Added stubs for things referenced in the docs and crosslinks to the stubs.  
- Added a glossary and some terms.
- Marked some todos.
- couple of typos.

### CODE REVIEW

File: docs/source/glossary.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+.. glossary::
+
+   Concept Model
+      A detailed description of an entire simulation model.  Concept models
+      use narrative, math equations, and diagrams to describe a system
+      we wish to study.
+
+   Entity Component System
+      One of the key software patterns used by ``vivarium`` to facilitate
+      the modular design of simulation models. In our modeling the entities are
```
[the remark this answers is not in the document]
Comment (collijk) [reply docs/source/glossary.rst]: feel free to make copy edits directly to the docs if you want.  Github's online editor is not bad.

File: docs/source/glossary.rst [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      A model that contains enough detail to accurately capture all
+      :term:`outcomes <Simulation Outcome>` of interest.
+
+   Validation
+      The process by which we determine whether a model is "correct",
+      meaning it makes logical sense, has the sort of behavior we expect,
+      and is sufficiently accurate for the intended modeling purposes.
+      Validation applies both to the :term:`concept model <Concept Model>`
+      and the simulation implementation of the model, though the criteria
+      we use for validation may be different.
```
[the remark this answers is not in the document]
Comment (collijk) [reply docs/source/glossary.rst]: I think what we've just done in the ltbi channel (and in the concept model meetings) is validation of the concept model.  It's systematically working through the model and proposed data sources to ensure they agree.  I'd like to see it also include the development of model assumptions and hypotheses and validation data to be tested on the simulation outputs.

[the remark this answers is not in the document]
Comment (aflaxman) [reply docs/source/glossary.rst]: I understand what you are saying now, but I think we might need a different name for this (eventually)

File: docs/source/glossary.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (alibow) [reply docs/source/glossary.rst]: Should we add cause models here? Also, should the description of a concept model note that cause models are contained within a concept model, or is that not exactly how we want to think about it?

File: docs/source/glossary.rst [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+   Verification
```
Comment (yongqx) [inline docs/source/glossary.rst]: Can we breakdown verification into three pieces: 
1. sub-minimal model: cause model plus necessary inputs and parameters/configuration.
2. minimal model: add risk factor and intervention component with all missing measures filled.
3. full model: more draws and adjustment of parameters if needed.

**Pull request merged**